tests/test_transaction.py

```diff
@@ -141,53 +158,202 @@ def test_transaction_void(self):
+        for line_item in line_items:
+            adjustments_to_refund.append({
+                'adjustment': line_item,
+                'quantity': line_item.quantity,
+                'prorate': False 
+            })
+        new_invoice = invoice.refund(adjustments_to_refund)
+
+        # Verify behavior of invoice line item refund
+        # - Creates a new invoice with adjustments that cancel out original invoice
+        # - Creates a new transaction object to be REFUND
+        self.assertEqual(len(mocurly.backend.transactions_backend.datastore), 2)
+        self.assertEqual(len(mocurly.backend.invoices_backend.datastore), 2)
+        self.assertEqual(len(mocurly.backend.adjustments_backend.datastore), 2)
+
+        for invoice_number, invoice in mocurly.backend.invoices_backend.datastore.items():
+            self.assertEqual(len(invoice['transactions']), 1)
+            if invoice_number == '1234':
+                original_invoice = invoice
+                self.assertEqual(invoice['transactions'][0], '1234')
+            else:
+                refund_invoice = invoice
+                self.assertNotEqual(invoice['transactions'][0], '1234')
+        self.assertEqual(original_invoice['total_in_cents'], -refund_invoice['total_in_cents'])
+
+        for adjustment_uuid, adjustment in mocurly.backend.adjustments_backend.datastore.items():
+            if adjustment['invoice'] == '1234':
+                original_adjustment = adjustment
+            else:
+                refund_adjustment = adjustment
+        self.assertEqual(original_adjustment['total_in_cents'], -refund_adjustment['total_in_cents'])
+        self.assertEqual(int(original_adjustment['quantity']), -int(refund_adjustment['quantity']))
+
+        for transaction_uuid, transaction in mocurly.backend.transactions_backend.datastore.items():
+            if transaction_uuid == '1234':
+                original_transaction = transaction
             else:
-                self.assertEqual(voided_transaction[k], v)
+                refund_transaction = transaction
+        self.assertEqual(original_transaction['action'], 'purchase')
+        self.assertEqual(original_transaction['status'], 'success')
+        self.assertEqual(original_transaction['voidable'], False)
+        self.assertEqual(original_transaction['refundable'], False)
+        self.assertEqual(refund_transaction['action'], 'refund')
+        self.assertEqual(refund_transaction['status'], 'success')
+        self.assertEqual(refund_transaction['voidable'], True)
+        self.assertEqual(refund_transaction['refundable'], False)
+
+        self.assertEqual(new_invoice.original_invoice().invoice_number, 1234)
 
-    def test_transaction_refund(self):
+
+    def test_transaction_void_via_invoice(self):
+        """Uses the invoice refund system to void a transaction."""
         self.assertEqual(len(mocurly.backend.transactions_backend.datastore), 0)
 
+        self.base_transaction_data['uuid'] = '1234'
```

Comment:
Can you consider making DRYer some of the test setups for the base transaction, invoice and base line?